---
title: "03_id_alga"
author: "Gulnara Tagridzhanova"
date: "20/03/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/path/to/repo/notebook")
```

**Summary:** After I recieved genomic assembly generated by Future Genomics from a lab culture, I identified the alga/ It wasn't a Trebouxia, as we expected, but Coccomyxa viridis (or a closely related strain). This is interesting in itself, since this alga has been reported from lichens that have Treboxia as the main photobiont.

## 1. NCBI blast evidence 
* Future Genomics generated a report, where they blasted all contigs from the algal sample against NCBI_nr (exact command is unknown), which came out as Coccomyxa


## 2. Phylogenomic tree
* Here I made a phylogenomic tree using reference transcriptomes from the literature and our algal strain
* In total, used 8 transcriptomes, 3 genomes, and 196 loci

#### Dataset construction
* NB: in this version of the analysis (unlike the previous version), I only used publicly available data in addition to the target genome. No unpublished data produced by the lab are included
* As references used:
  1. all publicly available Trebouxiophyceaea transcriptomes from JGI  
    * `aster`: Asterochloris from [JGI](https://phycocosm.jgi.doe.gov/Astpho2/Astpho2.home.html) (cite Armaleo et al. BMC Genomics, 2019)
    * `treba12`: Trebouxia from [JGI](https://phycocosm.jgi.doe.gov/TrebA12_1/TrebA12_1.home.html). (cite Greshake Tzovaras et al. GBE, 2020) **NB:** *might* not the same species as the one we have in cultures
    * `chlorel`: Chlorella sorokiana from [JGI](https://phycocosm.jgi.doe.gov/ChloDOE1412_1/ChloDOE1412_1.home.html) (cite Hovde et al. Algal Research, 2018)
    * `chvar`: [Chlorella variabilis](https://genome.jgi.doe.gov/portal/ChlNC64A_1/ChlNC64A_1.download.html) (Blanc et al. Plant Cell. 2010)
    * `auxeprot`: [Auxenochlorella protothecoides](https://phycocosm.jgi.doe.gov/Auxeprot1/Auxeprot1.home.html)(Gao et al. BMC Genomics. 2014)
    * `botrbrau`: [Botryococcus braunii](https://phycocosm.jgi.doe.gov/Botrbrau1/Botrbrau1.home.html) (Browne et al. Genome Announc. 2017)
    * `cosub3`: [Coccomyxa subellipsoidea](https://phycocosm.jgi.doe.gov/Cosub3/Cosub3.home.html) (Blanc et al. Genome Biol. 2012)
  2. Additional Trebouxiales from NCBI
    * `trebgel`: [Trebouxia gelatinosa](https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_000818905.1/). Isolated from Gyalolechia (former Caloplaca)
    * `trebus`: [Trebouxia sp. TZW2008, photobiont of Usnea](https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_002118135.1/) (Kono et al. BMC Genomics, 2020)
  3. As outgroup, `chleu`: [Chlamydomonas eustigma](https://phycocosm.jgi.doe.gov/Chleu1/Chleu1.home.html) (Hirooka et al. PNAS 2017)
  
#### BUSCO annotation
* This tree is based on BUSCO genes shared between all genomes/transcriptomes in our dataset. 
* Created a new folder and copied there BUSCO annotation of algal MAG, see `02_binning` )
```
mkdir -p analysis_and_temp_files/03_id_alga/busco/GTX0488_bin9/run_chlorophyta_odb10/
cp analysis_and_temp_files/02_binning/GTX0488_algal_MAG_busco/GTX0488_bin9/run_chlorophyta_odb10/full_table.tsv analysis_and_temp_files/03_id_alga/busco/GTX0488_bin9/run_chlorophyta_odb10/
```
* For `cosub3`, original file contained forward slash in the fasta headings. Replaced them using tr
```
tr -d '/' < /tsl/data/externalData/ntalbot/lichen_project/external_genomes/cosub3_transcripts.fasta  > /tsl/data/externalData/ntalbot/lichen_project/external_genomes/cosub3_fixed_transcripts.fasta 
```
* All of subsequent steps are implemented as a part of Snakemake pipeline (the code is in `analysis_and_temp_files/03_id_alga/Snakefile_busco_alg`). In brief:
  * I annotated reference genomes and transcriptomes
  * Selected BUSCO genes that are complete and single-copy in all 
  * BUSCO in its transcriptome mode, doesn't produce final BUSCO sequences, only a temp file with all 6 reading frames. To get the right sequences among the six, I ran a blastp search using proteins from Trebouxia as a bait
  * Concatenated sequences, aligned with MAFFT and trimmed with trimAL to remove positions present in <70% of organsims
  * Build a phylogeny with RAxML, using PROTGAMMAAUTO model

### Results:
The target genome is sister to Coccomyxa.

```{r, message = FALSE,fig.width=14,fig.height=6}
library(ape)
library(ggtree)
library(tidyverse)

tree<-read.tree("../analysis_and_temp_files/03_id_alga/busco/alignments/RAxML_bestTree.raxml_tree")
tree <- root(tree, outgroup = "chleu", resolve.root = TRUE)

##rename
old_names<-c("aster","auxeprot", "chvar","chlorel","chleu","botrbrau","GTX0488_bi", "trebus","treba12","trebgel","cosub3_fix")   
new_names<-c("Asterochloris glomerata, Trebouxiales (JGI)","Auxenochlorella protothecoides, Chlorellales (JGI)","Chlorella variabilis, Chlorellales (JGI)","Chlorella sorokiana, Chlorellales (JGI)","Chlamydomonas eustigma, Chlorophyceae (JGI, Outgroup)","Botryococcus braunii, Elliptochloris clade (JGI)","Coccomyxa viridis NJT Lab Strain","Trebouxia  sp. TZW2008, Trebouxiales (NCBI)","Trebouxia A1-2, Trebouxiales (NCBI)","Trebouxia gelatinosa, Trebouxiales (NCBI)","Coccomyxa subellipsoidea, Elliptochloris clade (JGI)")
df<-data.frame("old_names"=old_names,"new_names"=new_names)
tree$tip.label<-df[[2]][match(tree$tip.label, df[[1]])]
df_col<-data.frame("new_names"=tree$tip.label) %>% mutate(labels=ifelse(new_names=="Coccomyxa viridis NJT Lab Strain","our_strain","other")) 

p <- ggtree(tree,ladderize = TRUE,right = T,linewidth=0.2)  %<+% df_col + 
  geom_hilight(node=18, fill="seagreen", alpha=.2,type="roundrect",extend=1)+
  geom_tiplab(pch=16, size=3,aes(col=labels),show.legend = FALSE)+
  scale_color_manual(values=c("our_strain" = "red","other"="black"),)
p + xlim(0, 2)
ggsave("../results/phylogenomic_tree.svg",device="svg",height=2,width=7)
```


## 3. Phylogenetic tree based on ITS
* For better taxonomic resolution, I built a tree using ITS sequences

#### Dataset construction
* Retrieved ITS sequence from the algal assembly
* Used blast+-2.9.0 and SAMtools v1.12
```
blastn -query analysis_and_temp_files/03_id_alga/genbank_coccomyxa_its.fa -subject data/FG23004_01_flye-medaka_2xpilon.fasta -outfmt 6
samtools faidx data/FG23004_01_flye-medaka_2xpilon.fasta contig_22_pilon_pilon:1121628-1122410 > analysis_and_temp_files/03_id_alga/GTX0488_its.fa

```
* When blasted against NCBI, the best hit is Coccomyxa viridis (100% coverage, 98% identity)
* Manually selected Coccomyxa ITS sequences and one Elliptochloris from NCBI
* For each sequence, I identified its source (see `analysis_and_temp_files/03_id_alga/its_id.txt`). I included:
  * Whether it's isolated from a lichen or a non-lichen source. For several seqeunces, this information wasn't available
  * If from lichen, what lichen species was used
  * Based on this information, I tentatively separated lichen-derived sequences in 'photobionts' (if coming from lichens whose primary photobiont is known to be Coccomyxa), and 'epiphytes' (if coming from lichens with a different photobiont). For the information on the lichen photobiont, I mostly relied on Matumoto & Sanders (2002)

#### Aligned and built a tree
* Aligned with mafft v7.271,trimmed with trimal-1.2, built the tree with raxml-8.2.12
```
cat analysis_and_temp_files/03_id_alga/genbank_its.fasta analysis_and_temp_files/03_id_alga/GTX0488_its.fa > analysis_and_temp_files/03_id_alga/combined_its.fa
mafft --maxiterate 1000 --genafpair --thread 20 analysis_and_temp_files/03_id_alga/combined_its.fa > analysis_and_temp_files/03_id_alga/combined_its_aligned.fa
trimal -in analysis_and_temp_files/03_id_alga/combined_its_aligned.fa -out analysis_and_temp_files/03_id_alga/combined_its_aligned.phyl -gt 0.2 -phylip -keepheader

bash code/iqtree.sh analysis_and_temp_files/03_id_alga/combined_its_aligned.phyl 20
```
#### The sequence was recovered in the Coccomyxa viridis clade
```{r, message = FALSE,fig.width=30,fig.height=30}
library(ape)
library(ggtree)
library(tidyverse)
tree2<-read.tree("../analysis_and_temp_files/03_id_alga/combined_its_aligned.phyl.contree")
tree2 <- root(tree2, outgroup = "LC660445.1", resolve.root = TRUE)
tree2 <- as.polytomy(tree2, feature='node.label', fun=function(x) as.numeric(x) < 70)

##rename
lit<-read.delim2("../analysis_and_temp_files/03_id_alga/its_id.txt",header=F)
colnames(lit)<-c("sequence","name","Coccomyxa_role","Lichen.metagenomes","Photobiont","note","link")
lit<-lit %>% mutate(new=ifelse(Coccomyxa_role %in% c("non-photobiont-lichen","photobiont"),
                          paste(sequence,name,"from",Lichen.metagenomes),
                          paste(sequence,name)))
tree2$tip.label<-lit[[8]][match(tree2$tip.label, lit[[1]])]

#make a vector for color
df<-data.frame("new"=tree2$tip.label) %>% left_join(lit) %>% 
  mutate(labels=ifelse(grepl("contig",new),"strain from this paper",Coccomyxa_role))


p <- ggtree(tree2,ladderize = TRUE,right = T,linewidth=0.2) %<+% df + 
     geom_hilight(node=82, fill="steelblue", alpha=.2,type="roundrect",extend=0.32)+
  geom_tiplab(pch=16, aes(col=labels),size=3)+
  scale_color_manual(values=c("strain from this paper" = "red","non-photobiont-lichen"= "#d55e00", "photobiont" = "#009e73","unknown"="#696969","non-lichen"="black"))+theme(legend.text=element_text(size=4),legend.title=element_blank())
p + xlim(0, 0.5)
ggsave("../results/ITS_tree.svg",device="svg",width=9,height=9)
```


## 4. Conclusion: our algal strain is actually Coccomyxa viridis
* Coccomyxa viridens **has been reported before** from some lichens that have Trebouxia as the main photobiont
  * [Molins et al. 2021, Journal of Phycology](https://onlinelibrary.wiley.com/doi/full/10.1111/jpy.13140), reported C. viridens from Ramalina thalli using amplicon metabarcoding
  * [Miral et al. 2022, Environmental Microbiology Reports](https://ami-journals.onlinelibrary.wiley.com/doi/full/10.1111/1758-2229.13105), cultured C. viridens from Rhizocarpon thalli
  * [Faluaburu et al. 2019, Microorganisms](https://www.mdpi.com/2076-2607/7/7/203), reported C. viridens from Umbilicaria thalli using amplicon metabarcoding
  * [Cao et al. 2018, PhytoKeys](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5966503/), isolated a close relative, C. antarctica, from an Usnea thallus
* The C. viridis clade contains many lichen-isolated alga. The majority of them are epiphytic (i.e. not the primary photobiont of a given lichen). The clade also included sequences that likely come from true photobiont of Micarea spp. C. viridis SAG 216-4 was labelled in the NCBI as the photobiont of Peltigera aphtosa (likely, because it was cultured from a Peltigera thallus). However, I suspect this alga is actually an epiphyte, given that all other sequences of Peltigera aphtosa photobiont are recovered in a completely different clade 



